module/zfs/dmu.c

```diff
@@ -2022,6 +2022,23 @@ dmu_object_set_blocksize(objset_t *os, uint64_t object, uint64_t size, int ibs,
 	return (err);
 }
 
+int
+dmu_object_set_maxblkid(objset_t *os, uint64_t object, uint64_t maxblkid,
+    dmu_tx_t *tx)
+{
+	dnode_t *dn;
+	int err;
+
+	err = dnode_hold(os, object, FTAG, &dn);
+	if (err)
+		return (err);
+	rw_enter(&dn->dn_struct_rwlock, RW_WRITER);
+	dnode_new_blkid(dn, maxblkid, tx, B_FALSE);
+	rw_exit(&dn->dn_struct_rwlock);
+	dnode_rele(dn, FTAG);
+	return (err);
```

Comment:
nit: functionally it doesn't matter but it'd be nice to explicitly `return (0)` here, the other newly added crypto functions written in the same style arguably should do the same as well.

Reply:
I copied this from `dmu_object_set_nlevels()` and `dmu_object_set_blocksize()`. Do you want me to change these too?

Reply:
Yes, please do.  It looks like you got a little unlucky and happened to pick a non-typical example function.  The code isn't as consistent as one might like about this.